### DISCUSSION TITLE
Type declaration for Link causes type check error in IDE

### DISCUSSION THREAD

**Discussion by hannasage** [gatsbyjs/gatsby#35709] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Using Gatsby with TypeScript out of the box, I'm unable to use `Link` from `gatsby` as a JSX element.

![error](https://user-images.githubusercontent.com/4022304/169426118-fd087e2d-ae7f-4b62-86b0-17979e55317e.png)

### Reproduction Link

https://github.com/kevinhaube/dev-blog

### Steps to Reproduce

1. Clone repo and go to `main`
2. Load up project in IDE with TS checks enabled (InellijIDEA 2022.1 for me)
3. Open `index.tsx` and observe the error in the `NavItem` component.

### Expected Result

The type successfully passes checking as a `ReactNode` and the error does not appear.

### Actual Result

`Link` causes type error when IDE parses types

### Environment

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 Intel(R) Core(TM) i5-10600K CPU @ 4.10GHz
  Binaries:
    Node: 16.14.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.3.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1.0), Chromium (101.0.1210.39)
  npmPackages:
    gatsby: ^4.14.1 => 4.14.1 
    gatsby-plugin-mdx: ^3.14.0 => 3.14.0 
    gatsby-plugin-react-helmet: ^5.14.0 => 5.14.0 
    gatsby-plugin-styled-components: ^5.14.0 => 5.14.0 
    gatsby-source-filesystem: ^4.14.0 => 4.14.0
```

### Config Flags

n/a

**Comment by LekoArts** [comment, marked as answer]
You're using the outdated types, that's why

![image](https://user-images.githubusercontent.com/16143594/169481611-24cc5915-a513-4764-b7cc-6ba9f92f0dd5.png)

Update to

```json
    "@types/react": "^18.0.9",
    "@types/react-dom": "^18.0.4",
```

And it should be fine, at least when I tried it with your repo.

**Reply by hannasage** [reply]
Thanks! Will try this and report back

**Comment by ursaj** [comment]
Faced the same problem with `@types/react` and `@types/react-dom` of version `^18.3`. After several attempts I've found it's a problem of new typings. Rolled back to `~18.0` for typings and `^18.3` for implementations. Looks like it's working now.
